polkadot/runtime/parachains/src/paras_inherent/weights.rs

```diff
@@ -123,7 +131,8 @@ where
 	set_proof_size_to_tx_size(
 		<<T as Config>::WeightInfo as WeightInfo>::enter_variable_disputes(
 			statement_set.as_ref().statements.len() as u32,
-		),
+		)
+		.saturating_sub(<<T as Config>::WeightInfo as WeightInfo>::enter_empty()),
```

Comment:
Can't we perform the `saturating_sub` during weight generation, instead of the helpers during execution ?